colibri/src/icons.rs

```diff
@@ -261,68 +397,67 @@ pub async fn query_icon_remotely(
         _ => None
     } {
         if let Some(icon_bytes) = query_image_from_cdn(url).await {
-            let _ = tokio::fs::write(path.with_extension(extension), &icon_bytes)
-                .await
-                .map_err(|e| {
-                    error!("Unable to write {} to the file system due to {}", path.display(), e);
-                });
-            return;
+            return write_icon_to_file(&path, extension, &icon_bytes).await;
         }
-    };
+    }
 
-    let (icon_bytes, extension) = match extract_chain_id_and_address(&asset_id) {
-        Some((chain_id, address)) => {
-            match query_token_icon_and_extension(chain_id, &address, SMOLDAPP_BASE_URL).await {
-                Some((bytes, ext)) => (bytes, ext),
-                None => match coingecko.query_asset_image(&asset_id).await {
-                    Some(bytes) => (bytes, "png"),
-                    None => {
-                        debug!(
-                            "Icon not found for asset {}. Writing zero bytes file",
-                            asset_id
-                        );
-                        write_zero_bytes_file(path.as_path()).await;
-                        return;
+    // Parse asset identifier for EVM-based assets
+    if let Some(asset_info) = parse_asset_identifier(&asset_id) {
+        // Handle NFTs - only check for Uniswap V3 if we have a token ID
+        if let Some(token_id) = &asset_info.token_id {
+            if let Some(manager_address) = get_uniswap_nft_manager(asset_info.chain_id) {
+                if manager_address.to_string() == asset_info.contract_address {
```

Comment:
isn't it better to query the gloaldb here before checking the chain? This will trigger to every possible nft